### PR TITLE
[Console] Check for ErrorHandler classes

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 4.
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Console/Application.php [hunk 1]
```diff
@@ -41,8 +41,10 @@
 use Symfony\Component\Console\Output\ConsoleOutputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
-use Symfony\Component\Debug\ErrorHandler;
-use Symfony\Component\Debug\Exception\FatalThrowableError;
+use Symfony\Component\Debug\ErrorHandler as LegacyErrorHandler;
+use Symfony\Component\Debug\Exception\FatalThrowableError as LegacyFatalThrowableError;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+use Symfony\Component\ErrorHandler\Exception\FatalThrowableError;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Application.php]: ideally we would get rid of that one :)

Comment (fancyweb) [reply src/Symfony/Component/Console/Application.php]: We will (as we discussed on Slack). But let's fix the ErrorHandler support here, before another PR that remove the "fatal throwable errors" behavior.

File: src/Symfony/Component/Console/Application.php [hunk 2]
```diff
@@ -137,10 +139,10 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         };
         if ($phpHandler = set_exception_handler($renderException)) {
             restore_exception_handler();
-            if (!\is_array($phpHandler) || !$phpHandler[0] instanceof ErrorHandler) {
-                $debugHandler = true;
-            } elseif ($debugHandler = $phpHandler[0]->setExceptionHandler($renderException)) {
-                $phpHandler[0]->setExceptionHandler($debugHandler);
+            if (!\is_array($phpHandler) || (!$phpHandler[0] instanceof ErrorHandler && !$phpHandler[0] instanceof LegacyErrorHandler)) {
+                $errorHandlerHandler = true;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Application.php]: just $errorHandler?

**Review by nicolas-grekas — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @fancyweb.

**Pull request merged**